src/fpm_backend_output.f90

```diff
@@ -0,0 +1,182 @@
+!># Build Backend Progress Output
+!> This module provides a derived type `build_progress_t` for printing build status
+!> and progress messages to the console while the backend is building the package.
+!>
+!> The `build_progress_t` type supports two modes: `normal` and `plain`
+!> where the former does 'pretty' output and the latter does not.
+!> The `normal` mode is intended for typical interactive usage whereas
+!> 'plain' mode is used with the `--verbose` flag or when `stdout` is not attached
+!> to a terminal (e.g. when piping or redirecting `stdout`). In these cases,
+!> the pretty output must be suppressed to avoid control codes being output.
+
+module fpm_backend_output
+use iso_fortran_env, only: stdout=>output_unit
+use fpm_filesystem, only: basename
+use fpm_targets, only: build_target_ptr
+use fpm_backend_console, only: console_t
+use M_attr, only: attr, attr_mode

```

Comment:
What is the reason for using `M_attr` here and using explicit escape codes in the `fpm_backend_console` module? We could add the required escape codes in the console type and use them here so the backend output can rely on one type / module handling the color output.

Reply:
Good point, I agree. I started off using `M_attr` but then needed to create `fpm_backend_console` anyway. Yeah, having it all handled in one place is better.